GoRogue/MapGeneration/Region.cs

```diff
@@ -9,16 +10,61 @@
 
 namespace GoRogue.MapGeneration
 {
+    /// <summary>
+    /// Event fired when a region's Area is changed.
+    /// </summary>
+    [PublicAPI]
+    public class RegionAreaChangedEventArgs : EventArgs
+    {
+        /// <summary>
+        /// The previous Area
+        /// </summary>
+        public readonly PolygonArea OldValue;
+
+        /// <summary>
+        /// The Area to which we update
+        /// </summary>
+        public readonly PolygonArea NewValue;
+
+        /// <summary>
+        /// The Event Arguments for when a region's area is changed
+        /// </summary>
+        /// <param name="item">The region whose area is changing</param>
+        /// <param name="oldValue">The former value of the Area</param>
+        /// <param name="newValue">The new value of the Area</param>
+        public RegionAreaChangedEventArgs(Region item, PolygonArea oldValue, PolygonArea newValue)
```

Comment:
```suggestion
        public RegionAreaChangedEventArgs(PolygonArea oldValue, PolygonArea newValue)
```

`Region` parameter is unused in current implementation.